Move odd num_head in TransformerEncoder to slow_path

zrphercule opened the pull request:
Summary: odd nhead is not supported for masked softmax, therefore we just move it to use old slow_path

Test Plan: CI

Differential Revision: D38720086

erichan1 reviewed (approved):
Thanks for the fix for https://github.com/pytorch/pytorch/issues/83355!

erichan1 commented:
Failure is because we did not add odd nhead check to creating a NestedTensor. So when nhead is odd, we are sending a NestedTensor to slow path, which cannot accept NestedTensor. Best fix is probably to check if the first or any encoder layer in TransformerEncoder has odd nhead. Kind of ugly... but what can you do.

zrphercule2 committed:
Move odd num_head in TransformerEncoder to slow_path (#83483)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/83483

odd nhead is not supported for masked softmax, therefore we just move it to use old slow_path

Test Plan: CI

Reviewed By: erichan1

Differential Revision: D38720086

fbshipit-source-id: 75abd450d6b6525b4c84470e5d189f9c498db753
